docs/articles/documentation/reference/test-api/selector/addcustomdomproperties.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: docs
+title: Selector.addCustomDOMProperties Method
+permalink: /documentation/reference/test-api/selector/addcustomdomproperties.html
+---
+# Selector.addCustomDOMProperties Method
+
+Adds custom properties to the [selector](../../../guides/basic-guides/select-page-elements.md).
+
+Use `addCustomDOMProperties` to retrieve DOM element properties not included in the standard selector API.
+
+```text
+Selector().addCustomDOMProperties({
+    property1: fn1,
+    property2: fn2,
+    /* ... */
+}) → Selector
+```
+
+Parameter                     | Type     | Description
+----------------------------- | -------- | -----------
+`property1`, `property2`, ... | String   | Property names.
+`fn1`, `fn2`, ...             | Function | Functions that calculate property values. Executed on the client side in the browser.
+
+Functions that calculate property values (`fn1`, `fn2`, ...) take the following parameters.
+
+Parameter   | Type     | Description
+----------- | -------- | -----------
+`node`      | Object   | The DOM node.
+
+**Example**
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('Check Label HTML', async t => {
+    const label = Selector('label').addCustomDOMProperties({
+        innerHTML: el => el.innerHTML
+    });
+
+    await t.expect(label.innerHTML).contains('input type="checkbox" name="remote"');
+});
+```
+
+If you use TypeScript, declare a Selector interface extension with your custom properties:
+
+<!--
+The `redcarpet` library ignores `ts` code and renders it as a plain text without code highligting.
+We can't use `js` here too. It's rendered wrong because of the type casting syntax. `csharp` looks
```

Comment:
here too?

Reply:
```suggestion
You cannot use `js` here either. It is rendered incorrectly because of the type casting syntax. `csharp` looks
```

Reply:
This comment is not displayed for the users.